src/node.cc

```diff
@@ -2752,28 +2752,12 @@ static void EnvGetter(Local<Name> property,
   if (property->IsSymbol()) {
     return info.GetReturnValue().SetUndefined();
   }
-#ifdef __POSIX__
   node::Utf8Value key(isolate, property);
-  const char* val = getenv(*key);
-  if (val) {
-    return info.GetReturnValue().Set(String::NewFromUtf8(isolate, val));
-  }
-#else  // _WIN32
-  node::TwoByteValue key(isolate, property);
-  WCHAR buffer[32767];  // The maximum size allowed for environment variables.
-  DWORD result = GetEnvironmentVariableW(reinterpret_cast<WCHAR*>(*key),
-                                         buffer,
-                                         arraysize(buffer));
-  // If result >= sizeof buffer the buffer was too small. That should never
-  // happen. If result == 0 and result != ERROR_SUCCESS the variable was not
-  // not found.
-  if ((result > 0 || GetLastError() == ERROR_SUCCESS) &&
-      result < arraysize(buffer)) {
-    const uint16_t* two_byte_buffer = reinterpret_cast<const uint16_t*>(buffer);
-    Local<String> rc = String::NewFromTwoByte(isolate, two_byte_buffer);
-    return info.GetReturnValue().Set(rc);
+  char buffer[32767];
+  int ret = uv_os_getenv(*key, buffer, arraysize(buffer));
```

Comment:
Shouldn't the final argument just be `sizeof(buffer)`?

Reply:
After reading the [manual](http://docs.libuv.org/en/v1.x/misc.html#c.uv_os_setenv) I figured it should be 🤦‍♂️ 
```
size_t buf_size = sizeof(buffer);
uv_os_getenv(*key, buffer, &buf_size);
```